### PR TITLE
feat: add Header

### PR DESCRIPTION
**Pull request by Souheil-Yazji**
Will probably refactor commit structure to seperate Header from Auth components. Draft PR made to track diffs.

Update: branch refactored to reflect only Header feature.

### CODE REVIEW

File: shopify-client/src/App.vue [hunk 1]
```diff
@@ -16,16 +15,18 @@
...
-  color: #42b983;
-}
+
 </style>
+
+<script>
+import Header from "@/components/Header";
+export default {
+  components: {
+    "Header": Header}
```
Comment (dershao) [inline shopify-client/src/App.vue:30]: _formatting_: put closing brace on the next line

File: shopify-client/src/components/Header.vue [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        <!-- Right aligned nav items -->
+        <b-navbar-nav class="ml-auto">
+          <b-nav-item to="/app/about">About</b-nav-item>
+          <b-nav-item to="/app/login">Login</b-nav-item>
+          <b-nav-item to="/app/signup">Sign Up</b-nav-item>
+        </b-navbar-nav>
+
+      </b-collapse>
+    </b-navbar>
+  </div>
```
Comment (dershao) [inline shopify-client/src/components/Header.vue:29]: _Nitpick_: unnecessary div (?)

File: shopify-client/src/components/Header.vue [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+      <b-collapse id="nav-collapse" is-nav>
+        <b-navbar-nav>
+          <b-nav-form>
+            <b-form-input placeholder="Search"></b-form-input>
+          </b-nav-form>
+        </b-navbar-nav>
+      </b-collapse>
+
+        <b-navbar-nav class="ml-auto">
```
Comment (dershao) [inline shopify-client/src/components/Header.vue:15]: _nitpick_: align the <b- > tags

**Review by dershao — CHANGES_REQUESTED** [review]
image 👍

File: shopify-client/src/components/Header.vue [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  </div>
+</template>
+
+<script>
+export default {
+  name: "Header"
+}
+</script>
+
+<style scoped>
```
Comment (DwangDistracted) [inline shopify-client/src/components/Header.vue:38]: Should we leave empty style tags?

**Review by DwangDistracted — APPROVED** [review]

File: shopify-client/src/components/Header.vue (hunk 4 above)
Comment (mpfan) [reply shopify-client/src/components/Header.vue:38]: should be removed if no styles inside

**Review by mpfan — CHANGES_REQUESTED** [review]

File: shopify-client/src/components/Header.vue (hunk 4 above)
Comment (Weebian) [reply shopify-client/src/components/Header.vue:38]: Leave it for the future, most likely will add styling

File: shopify-client/src/App.vue [hunk 5]
```diff
@@ -1,10 +1,9 @@
 <template>
   <div id="app">
     <div id="nav">
-      <router-link to="/">Home</router-link> |
-      <router-link to="/about">About</router-link>
```
Comment (mpfan) [inline shopify-client/src/App.vue:5]: I think you need router-link and router-view for the router to work. https://router.vuejs.org/api/#router-link

File: shopify-client/src/components/Header.vue [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+          <b-navbar-brand to="/app">MiniShopify</b-navbar-brand>
+        </b-navbar-nav>
+
+      <b-collapse id="nav-collapse" is-nav>
+        <!-- Right aligned nav items -->
+        <b-navbar-nav class="ml-auto">
+          <b-nav-item to="/app/about">About</b-nav-item>
+          <b-nav-item to="/app/login">Login</b-nav-item>
+          <b-nav-item to="/app/signup">Sign Up</b-nav-item>
+        </b-navbar-nav>
```
[the remark this answers is not in the document]
Comment (Weebian) [reply shopify-client/src/components/Header.vue:25]: /shops/list not /shops

File: shopify-client/src/components/Header.vue [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+      </b-collapse>
+
+        <b-navbar-nav class="ml-auto">
+          <b-navbar-brand to="/app">MiniShopify</b-navbar-brand>
+        </b-navbar-nav>
+
+      <b-collapse id="nav-collapse" is-nav>
+        <!-- Right aligned nav items -->
+        <b-navbar-nav class="ml-auto">
+          <b-nav-item to="/app/about">About</b-nav-item>
```
Comment (mpfan) [inline shopify-client/src/components/Header.vue:22]: You can wrap `<router-link>` inside `<b-nav-item`>.

**Review by mpfan — CHANGES_REQUESTED** [review]

File: shopify-client/src/components/Header.vue (hunk 2 above)
Comment (Weebian) [reply shopify-client/src/components/Header.vue:29]: it is necessary

Comment (mpfan) [reply shopify-client/src/components/Header.vue:29]: Yes since we have `<b-navbar>` wrapping everything already.

**Review by mpfan — CHANGES_REQUESTED** [review]

File: shopify-client/src/App.vue (hunk 5 above)
Comment (dershao) [reply shopify-client/src/App.vue:5]: `<b-nav-item>` is sufficient for navigating to different views.

File: shopify-client/src/components/Header.vue (hunk 4 above)
Comment (Souheil-Yazji) [reply shopify-client/src/components/Header.vue:38]: Leaving for future reference.

File: shopify-client/src/components/Header.vue (hunk 2 above)
Comment (Souheil-Yazji) [reply shopify-client/src/components/Header.vue:29]: Div left for styling/doc structure, may use later.

**Review by mpfan — CHANGES_REQUESTED** [review]
Add a button to create shop

**Comment by Souheil-Yazji** [comment]
> Add a button to create shop

Added a button just to make development easier but I'm not a fan of the styling. I intend to make the "Shop" nav item a dropdown with a MouseOver and onClick Events. Intended functionality would be:
onClick -> shops/list
MouseOver -> shows dropdown containing all shop related functionality

Let me know what you guys think but essentially, the objective is to keep the header de-cluttered and clean while keeping relating context items localized.

File: shopify-client/src/views/ShopDetails.vue [hunk 8]
```diff
@@ -85,7 +87,6 @@
                 this.$router.push(`/app/shops/${this.id}/update`);
             }
         },
-        props: ["id"],
```
Comment (mpfan) [inline shopify-client/src/views/ShopDetails.vue:88]: I want to keep the path variable.

**Review by mpfan — CHANGES_REQUESTED** [review]

File: shopify-client/src/views/ShopDetails.vue (hunk 8 above)
Comment (mpfan) [reply shopify-client/src/views/ShopDetails.vue:88]: https://medium.com/@fullsour/when-should-you-use-path-variable-and-query-parameter-a346790e8a6d. Basically if you want to identify a resource you want to use path variable.

Comment (Souheil-Yazji) [reply shopify-client/src/views/ShopDetails.vue:88]: reverted

**Review by mpfan — APPROVED** [review]

**Review by Weebian — APPROVED** [review]
+1

**Review by dershao — APPROVED** [review]

**Pull request merged**